File: Algoritmica/Actividades/Actividad-I/Codigos-I/Tema1_3_Analisis/Tema1_22.cpp
#include <iostream>
#include <ctime>
using namespace std;
const int MAXN = 10000000;

int busca(int v[], int x)
{
    for (int i = 0; i <= MAXN; i++)
        if (v[i] == x)
            return i;
    return -1;
}

int main()
{
    int n, x;
    static int v[MAXN + 1]; /* Vector donde buscar y su talla */

    for (int i = 0; i <= MAXN; i++)
        v[i] = i;

    //cout << "Dato a buscar: "<<endl; cin >> x;
    puts("MEJOR CASO");
    clock_t begin = clock();
    x = 0;
    n = busca(v, x);
    clock_t end = clock();
    double elapsed_secs = double(end - begin)/CLOCKS_PER_SEC;
    cout << "Posicion de " << x << ": " << n << endl;
    cout << "Este algoritmo requirio " << elapsed_secs << " segundos\n"
         << endl;

    puts("CASO MEDIO");
    begin = clock();
    x = MAXN / 2;
    n = busca(v, x);
    end = clock();
    elapsed_secs = double(end - begin) / CLOCKS_PER_SEC;
    cout << "Posicion de " << x << ": " << n << endl;
    cout << "Este algoritmo requirio " << elapsed_secs << " segundos\n"
         << endl;

    puts("CASO PEOR");
    begin = clock();
    x = MAXN;
    n = busca(v, x);
    end = clock();
    elapsed_secs = double(end - begin) / CLOCKS_PER_SEC;
    cout << "Posicion de " << x << ": " << n << endl;
    cout << "Este algoritmo requirio " << elapsed_secs << " segundos\n"
         << endl;

    return 0;
}

File: Algoritmica/Actividades/Actividad-I/Codigos-I/Tema1_3_Analisis/Tema1_22.py
from numpy import arange
import time
MAXN = 10000000


def busca(v, MAXN, x):  # busca x en v[0...n-1]
    for i in range(MAXN+1):
        if v[i] == x:
            return i
    return -1


v = arange(MAXN+1)

#x=int(input("Dato a buscar: "))

print("MEJOR CASO")
inicio = time.time()
i = busca(v, MAXN, 0)
final = time.time()
print("Posición de %d: %d\n" % (0, i))
print("Este algoritmo requirió %10.7f segundos\n\n" % (final-inicio))

print("CASO MEDIO")
inicio = time.time()
i = busca(v, MAXN, MAXN//2)
final = time.time()
print("Posición de %d: %d\n" % (MAXN//2, i))
print("Este algoritmo requirió %10.7f segundos\n\n" % (final-inicio))

print("PEOR CASO")
inicio = time.time()
i = busca(v, MAXN, MAXN)
final = time.time()
print("Posición de %d: %d\n" % (MAXN, i))
print("Este algoritmo requirió %10.7f segundos\n\n" % (final-inicio))
